content/bash-scripting-basics.md

```diff
@@ -0,0 +1,158 @@
+echo "Your favorite color is: $favoriteColor"
+```
+
+At this point, you should have a working script that asks the user for their name and their favorite color and then reacts to this information by outputting a sentence that reflects this information. The full script at this point should look like this:
+
+```bash
+#!/usr/bin/env bash
+echo "What is your first name?"
+read firstName
+echo "Your first name is: $firstName"
+echo "What is your favorite color?"
+read favoriteColor
+echo "Your favorite color is: $favoriteColor"
+```
+
+### If-Then Statements
+
+Sometimes we want different things to happen depending on what inputs/variables are stored. For instance: if a file exists, then do *a*, but if the file does not exist, then do *b*. Or if a number is higher than *x* then do *a*, but if the number is lower than *x*, then do *b*. We can do this with an `if` statement.
+
+For this, we will be asking the user to input a number. If the number is higher than *x* we will do *a*, but if the number is lower than *x*, we will do *b*.
+
+Add a new line to your script. We will be making the target number 23, but you can pick any number you want. We will be putting the target number into a variable called `targetNumber`. Add the following line to your script:
+
+```bash
+targetNumber=23
+```
+
+We will now ask the user to input a number, and store that number to a variable called `guess`. Add the following lines to your script:
+
+```bash
+echo "Guess the number:"
+read guess
+```
+
+We now need to create and `if` statement to compare the `targetNumber` to the user input `guess` variable. The script will output different sentences depending on if the user is too low, too high, or if they guess correctly. Add the following lines to your script:
+
+```bash
+if [ $guess == $targetNumber ]
+then
+  echo "You guessed the correct number!"
+elif [ $guess -lt $targetNumber ]
+then
+  echo "You guessed lower than the target number."
+elif [ $guess -gt $targetNumber ]
+then
+  echo "You guessed higher than the target number."
+else
+  echo "Something went wrong."
+fi
+```
+
+If you run the script, it will ask the user for their name and their favorite color. It will then ask the user to input a number. After the user inputs the number, depending on if it was lower, higher or correct, the script will dynamically adapt to the user input.
+
+## Conclusion
+
+BASH scripting can be very useful in many different cases. I use scripting to automatically create a new work note with the current date as the title of the note document. I also have scripts to open certain applications or change my desktop theme. I have developed scripts to convert audio files to a specific format, or fill out PDF documents quickly. I am still learning myself, so don't get discouraged or frustrated. Try to keep at it and as you make mistakes, you will learn for the next script you write. There are so many resources to help with your project. I have listed a few places to help with this below.
```

Comment:
> Try to keep at it and as you make mistakes,

I think this is a bit of a run-on. Chopping it into two sentences might work better:

> Try to keep at it. As you make mistakes, you will learn for the next script you write.